docs/machine-learning/tutorials/object-detection-onnx.md

```diff
@@ -0,0 +1,713 @@
+
+    [!code-csharp [AssetDefinition](~/machinelearning-samples/samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx/ObjectDetectionConsoleApp/Program.cs#L17-L21)]
+
+Add a new directory to your project to store your input data and prediction classes.
+
+In **Solution Explorer**, right-click the project, and then select **Add** > **New Folder**. When the new folder appears in the Solution Explorer, name it "DataStructures".
+
+Create your input data class in the newly created *DataStructures* directory.
+
+1. In **Solution Explorer**, right-click the *DataStructures* directory, and then select **Add** > **New Item**.
+1. In the **Add New Item** dialog box, select **Class** and change the **Name** field to *ImageNetData.cs*. Then, select the **Add** button.
+     
+    The *ImageNetData.cs* file opens in the code editor. Add the following `using` statement to the top of *ImageNetData.cs*:
+
+    [!code-csharp [ImageNetDataUsings](~/machinelearning-samples/samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx/ObjectDetectionConsoleApp/DataStructures/ImageNetData.cs#L1-L4)]
+
+    Remove the existing class definition and add the following code for the `ImageNetData` class to the *ImageNetData.cs* file:
+    
+    [!code-csharp [ImageNetDataClass](~/machinelearning-samples/samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx/ObjectDetectionConsoleApp/DataStructures/ImageNetData.cs#L8-L23)]
+
+    `ImageNetData` is the input image data class and has the following <xref:System.String> fields:
+
+    - `ImagePath` contains the path where the image is stored.
+    - `Label` contains the name of the file.
+
+    Additionally, `ImageNetData` contains a method `ReadFromFile` which loads multiple image files stored in the `imageFolder` path specified and returns them as a collection of `ImageNetData` objects.
+
+Create your prediction class in the *DataStructures* directory.
+
+1. In **Solution Explorer**, right-click the *DataStructures* directory, and then select **Add** > **New Item**.
+1. In the **Add New Item** dialog box, select **Class** and change the **Name** field to *ImageNetPrediction.cs*. Then, select the **Add** button.
+
+    The *ImageNetPrediction.cs* file opens in the code editor. Add the following `using` statement to the top of *ImageNetPrediction.cs*:
+
+    [!code-csharp [ImageNetPredictionUsings](~/machinelearning-samples/samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx/ObjectDetectionConsoleApp/DataStructures/ImageNetPrediction.cs#L1)]
+
+    Remove the existing class definition and add the following code for the `ImageNetPrediction` class to the *ImageNetPrediction.cs* file:
+
+    [!code-csharp [ImageNetPredictionClass](~/machinelearning-samples/samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx/ObjectDetectionConsoleApp/DataStructures/ImageNetPrediction.cs#L5-L9)]
+
+    `ImageNetPrediction` is the prediction data class and has the following `float[]` field:
```

Comment:
I suggest moving the grid image into this section, to break up the code a bit, and also to illustrate how the ImagePrediction data is encoded.

We could talk about the expected dimensions of the ImagePrediction float[] array too